supplementary/envelopes_analysis/field_envelopes.py

```diff
@@ -103,9 +105,13 @@
     pc = axs[0, i].pcolormesh(e2d, t2d, pulse_wigner, cmap='RdBu', vmin=-1, vmax=1)
     if i == len(envelope_types) - 1: fig.colorbar(pc, ax=axs[0, i], shrink=0.92, fraction=0.05)
 
-    pc = axs[1, i].pcolormesh(e2d, t2d, pulse_wigner, cmap='Blues', norm='log', vmin=1e-5, vmax=1)
+    # Use log scale for the colormap
+    # https://matplotlib.org/stable/users/explain/colors/colormapnorms.html#logarithmic
+    lognorm = matplotlib.colors.LogNorm(vmin=1e-5, vmax=1)
+    # Plot negative and positive parts separately in blue and red, respectively
+    pc = axs[1, i].pcolormesh(e2d, t2d, pulse_wigner, cmap='Blues', norm=lognorm)
     if i == len(envelope_types) - 1: fig.colorbar(pc, ax=axs[1, i], shrink=0.92, fraction=0.05)
-    pc = axs[1, i].pcolormesh(e2d, t2d, -pulse_wigner, cmap='Reds', norm='log', vmin=1e-5, vmax=1)
```

Comment:
I needed to change this since passing `norm='log'` did not work with an older matplotlib version that gets installed with Python 3.7

Reply:
Ok, nice solution.